go/kex2/provisioner.go

```diff
@@ -149,6 +154,9 @@ func (p *provisioner) runProtocolWithCancel() (err error) {
 		p.canceled = true
 		return ErrCanceled
 	case err = <-ch:
+		if _, ok := err.(rpc.CanceledError); ok && !p.helloReceived {
+			return ErrHelloTimeout
```

Comment:
does this error export properly over RPC? We'll likely need it for the desktop client, I'm guessing.